### PR TITLE
fix: update policy step with self approval flag

### PR DESCRIPTION
**Pull request by utsav14nov**

### CODE REVIEW

File: gotocompany/guardian/v1beta1/guardian.proto [hunk 1]
```diff
@@ -939,6 +939,7 @@ message Policy {
     string approve_if = 6;
     repeated string approvers = 7;
     string rejection_reason = 8;
+    bool dont_allow_self_approval = 9;
```
Comment (anjali9791) [inline gotocompany/guardian/v1beta1/guardian.proto:942]: would restrict_self_approval be a better name?

**Review by Gadigeppa-J — APPROVED** [review]
lgtm

File: gotocompany/guardian/v1beta1/guardian.proto (hunk 1 above)
Comment (Gadigeppa-J) [reply gotocompany/guardian/v1beta1/guardian.proto:942]: IMO, dont_allow_self_approval is more appropriate. 
'restrict_self_approval' implies reducing the scope of self-approval but not fully eliminating it.
One more options is - 'disable_self_approval'

Comment (rahmatrhd) [reply gotocompany/guardian/v1beta1/guardian.proto:942]: how about `block_self_approval` or `deny_self_approval`?

Comment (utsav14nov) [reply gotocompany/guardian/v1beta1/guardian.proto:942]: I understand `deny_self_approval` looks better in terms of technical language.
But we came up with `dont_allow_self_approval`  from the just opposite of `allow_self_approval`  as to maintain backward compatibility.
I think this name also gives the proper definition that self approval is not allowed and is fully eliminated. WDYT?

**Review by rahmatrhd — APPROVED** [review]

**Pull request merged**